website/docs/chapter-03/05_advanced_importing.md

```diff
@@ -0,0 +1,197 @@
+
+
+Some keyword libraries have the option to be configured to change their behavior, which may also change the available keywords they offer.
+
+
+
+## 3.5.1 Importing Hierarchies
+
+::::lo[Learning Objectives]
+
+:::K2[LO-3.5.1]
+
+Understand how transitive imports of resource files and libraries work.
+
+:::
+
+::::
+
+Let's assume the following libraries and resource files shall be used:
+- **Library**    `A`
+- **Library**    `B`
+- **Library**    `Operating System`
+- **Resource**    `tech_keywordsA.resource`
+- **Resource**    `tech_keywordsB.resource`
+- **Resource**    `variables.resource`
+- **Resource**    `functional_keywords.resource`
+
+The respective files could look like this:
+
+**tech_keywordsA.resource:**
+```robotframework
+*** Settings ***
+Library    A
+Library    Operating System
+```
+
+**tech_keywordsB.resource:**
+```robotframework
+*** Settings ***
+Library    B
+Resource    variables.resource
+```
+
+**functional_keywords.resource:**
+```robotframework
+*** Settings ***
+Resource    tech_keywordsA.resource
+Resource    tech_keywordsB.resource
+```
+
+**suite.robot:**
+```robotframework
+*** Settings ***
+Resource    functional_keywords.resource
+```
+
+In this case, the suite `suite.robot` has access to all keywords from all keyword libraries, as well as all variables and user keywords from all resource files.
+With this transitive importing it is possible to organize user keywords and imports of libraries in a hierarchical way.
+
+It shall be avoided to create circular imports, where `A.resource` imports `B.resource` and `B.resource` imports `A.resource`.
+
+It should be avoided to import the same library in different places multiple times.
+If the exact same library with the same configuration (see the next section) is imported again, it will be ignored because Robot Framework already has it in its catalog.
+However, if the library is imported with different configurations, it may be imported multiple times, but depending on the library’s internal behavior, the new configuration may have no effect on the existing keywords, or other side effects may occur.
+
+
+Therefore, the recommendation is to import libraries only in one resource file with one configuration and use that import file in all places where the library is needed to make its keywords available.
+
+
+
+## 3.5.2 Library Configuration
+
+::::lo[Learning Objectives]
+
+:::K3[LO-3.5.2]
+
+Be able to configure a library import using arguments.
+
+:::
+
+::::
+
+Some libraries offer or need additional configuration to change their behavior or make them work.
+This is typically global behavior like internal timeouts, connection settings to systems, or plugins that should be used.
+
+If this is possible, the library documentation will have an `Importing` section directly before the list of keywords.
+It is strongly recommended to have all these possible arguments to the library itself defined with default values;
+however, that is not always possible.
```

Comment:
This phrasing makes this feel out of scope for this syllabus. The reader is expected to know how to import libraries correctly, not how to create a library with import parameters. Therefore, the reader shouldn't need to be aware of what are the strong recommendations for library development (i.e. to add default values to possible arguments), but to know how to identify if a library requires mandatory arguments and how to import a library with arguments.